WordPress/src/main/java/org/wordpress/android/ui/domains/management/usedomain/composable/UseDomainScreen.kt

```diff
@@ -0,0 +1,249 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Devices
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import org.wordpress.android.R
+import org.wordpress.android.ui.compose.components.MainTopAppBar
+import org.wordpress.android.ui.compose.components.NavigationIcons
+import org.wordpress.android.ui.compose.theme.AppTheme
+
+@Composable
+fun UseDomainScreen(
+    onNewDomainCardSelected: () -> Unit,
+    onExistingDomainCardSelected: () -> Unit,
+    onBackPressed: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val contentScrollState = rememberScrollState()
+    Scaffold(
+        modifier = modifier,
+        topBar = {
+            val elevation = animateDpAsState(
+                targetValue = if (contentScrollState.value == 0) 0.dp else 4.dp,
+                label = "AppBarElevation"
+            )
+            MainTopAppBar(
+                title = stringResource(id = R.string.use_domain_screen_title),
+                navigationIcon = NavigationIcons.BackIcon,
+                elevation = elevation.value,
+                onNavigationIconClick = onBackPressed
+            )
+        },
+        content = {
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .background(MaterialTheme.colors.surface)
+                    .padding(it)
+            ) {
+                Column(
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .verticalScroll(contentScrollState)
+                        .padding(16.dp)
+                ) {
+                    ScreenHeader()
+                    ScreenDescription()
+                    DomainCards(onNewDomainCardSelected, onExistingDomainCardSelected)
+                    DiscountNotice()
+                }
+            }
+        }
+    )
+}
+
+@Composable
+private fun DomainCards(
+    onNewDomainCardSelected: () -> Unit,
+    onExistingDomainCardSelected: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Box(modifier = modifier.padding(top = 24.dp)) {
+        if (LocalConfiguration.current.orientation == Configuration.ORIENTATION_PORTRAIT) {
+            Column {
+                NewDomainCard(onNewDomainCardSelected)
+                ExistingDomainCard(onExistingDomainCardSelected, modifier = Modifier.padding(top = 16.dp))
+            }
+        } else {
+            Row {
+                NewDomainCard(onNewDomainCardSelected, modifier = Modifier.weight(1f))
+                Spacer(modifier = Modifier.width(16.dp))
+                ExistingDomainCard(onExistingDomainCardSelected, modifier = Modifier.weight(1f))
+            }
+        }
```

Comment:
I think this is fine as it is, but just wanted to note that we could also achieve the necessary spacing via the `Row`'s and `Column`'s respective parameters `horizontalArrangement` and `verticalArrangement`, e.g.:

```suggestion
        val arrangement = Arrangement.spacedBy(16.dp)

        if (LocalConfiguration.current.orientation == Configuration.ORIENTATION_PORTRAIT) {
            Column (verticalArrangement = arrangement) {
                NewDomainCard(onNewDomainCardSelected)
                ExistingDomainCard(onExistingDomainCardSelected)
            }
        } else {
            Row (horizontalArrangement = arrangement) {
                NewDomainCard(onNewDomainCardSelected, modifier = Modifier.weight(1f))
                ExistingDomainCard(onExistingDomainCardSelected, modifier = Modifier.weight(1f))
            }
        }
```

Reply:
Wow, `Today I Learned`! Thank you for the suggestion 🤩